src/option.c

```diff
@@ -8170,3 +8195,22 @@ option_set_callback_func(char_u *optval UNUSED, callback_T *optcb UNUSED)
     return FAIL;
 #endif
 }
+
+#if defined(FEAT_EVAL) || defined(PROTO)
+    static void
+didset_options_sctx(int opt_flags, char **buf)
+{
+	int i;
+	int idx;
+
+	for (i = 0; ; ++i)
```

Comment:
Can you move the declaration of `i` into the for loop?

Reply:
done